### ISSUE TITLE
disconnect when re-entering the lobby

### ISSUE THREAD

**Issue by NL0bP** [NL0bP/Archeage-Server-emulator#2]
disconnect when re-entering the lobby

**Comment by NL0bP** [comment]
Yes, it's unexpected .... But there are 11 forks, for example [here] (https://github.com/veraalex97/Archeage-Server-emulator-or-files).

**Comment by JeyFrey** [comment]
Hey I also keep downloading all new files and stages of development, so if all stuff gets deleted, I still have a few backups.

**Comment by Kalle801** [comment]
Hello again :) I try to get into the work with the newst files but i cant even load the characters. It is like the server does not read the tables.

What do i have to do with the: archeage_world Database which i can not find ?

**Comment by NL0bP** [comment]
For LoginServer, the database should be called archeage. The database must have tables accounts and characters.

**Comment by NL0bP** [comment]
Let me remind you: the password is encrypted by MD5 and written to the token field.

**Comment by NL0bP** [comment]
GameServer does not use any database yet, so it's like it's written in the config.

**Comment by Kalle801** [comment]
i got it all like this but i use your filesonly, do i need other files also?

**Comment by sxeroot** [comment]
@Kalle801 localhost or 127.0.0.1 run mysql servers usualy have ssl off unless configured otherwise, loginserver tries to connect with ssl you need to add `SslMode=none` to the connection string in the Settings.cs file of the loginserver

**Comment by Kalle801** [comment]
hm i dont know ehere to add it. Im still learning c#

**Comment by sxeroot** [comment]
@Kalle801 

**Loginserver files Settings.cs line 24

looks like** 

`string connection = "server=" + Settings.Default.DataBase_Host + ";user=" + Settings.Default.DataBase_User + ";database=" + Settings.Default.DataBase_Name + ";port=" + Settings.Default.DataBase_Port + ";password=" + Settings.Default.DataBase_Password + ";";`

**change to**

`string connection = "server=" + Settings.Default.DataBase_Host + ";user=" + Settings.Default.DataBase_User + ";database=" + Settings.Default.DataBase_Name + ";port=" + Settings.Default.DataBase_Port + ";password=" + Settings.Default.DataBase_Password + ";SslMode=none";`

**Comment by Kalle801** [comment]
Got it! Thank you :)

**Comment by Kalle801** [comment]
@NL0bP Would you be so kind and upload the client u use? With the ArcheRage client i cant do anything.
Or are you using another client database?

**Comment by NL0bP** [comment]
This is all that is at the moment. There is no one, nothing can be done. You can only see the world.

**Comment by Kalle801** [comment]
If i got an ArcheAge mysql database (full) not encrypted. Would that help?

**Comment by Rain0Ash** [comment]
I think yes. More - no less, come in handyю

**Comment by NL0bP** [comment]
Yes, if you can give a link to it.

**Comment by Rain0Ash** [comment]
Perhaps not in the subject, but as far as I can remember, in the early versions of AA there was no protection for the packages. It remains to find a beta client or early patches 1.x. But it's just, for the sake of information.

**Comment by JeyFrey** [comment]
We need to find out how the database can be decrypted, I am just extracting the game_pak (still loading) from the 3.0.3.0 version and found some websites with hints how to decrypt it. Will keep you updated, but I literally just started learning programming, so dont expect anything.

**Comment by Kalle801** [comment]
i got the client (game_pak) editor if someone needs it.

**Comment by JeyFrey** [comment]
Unpacking the game_pak gives me lots of data, including compact.sqlite - but it seems one needs a compact.sqlite3 to do anything further instead, so how do you convert a .sqlite to .sqlite3?

**Comment by sxeroot** [comment]
@JeyFrey you dont, or more like you cant with that one since it needs to be decrypted first that is the issue the old databases werent encrypted cant remmeber the point of when they started encrypting them but the 2013 leaked files had a sqlite3 file that had alot of things maybe as a base could be used link below

https://anonfile.com/2fw6N7fcba/compact.sqlite3

i believe its more or less the same db provided in the other opened issue as plain sql there

**Comment by JeyFrey** [comment]
@sxeroot I see. I thought from reading this thread post, that it should still work if I find any way to convert it into the right sqlite3:  http://forum.ragezone.com/f857/tools-archeage-db-converter-pack-1040661/#post8281281

So what we basically need is a way to encrypt the new compact.sqlite as well as a way to create + delete characters, right?

**Comment by sxeroot** [comment]
@JeyFrey i did try that method described in ragezone that you linked yesterday evening tho it was getting late and didnt get any promising results so i gave up, that there takes in an sqlite3 file can change it to take sqlite instead and see if it decrypts it will do that later today and post results.

as for the rest if someone is able to decrypt it would be awesome since we would get all the structures used by the game however they arent really nessesary the database can be recreated in theoretical way like the character table we know what the game will need to store more or less on character data so we can play around with it ofcourse this will take longer to do, tho i still believe the old database can be atleast used as base and newer stuff added to it because even tho its been like 5 years probably not much has changed in the main structure of the database.

**Comment by sxeroot** [comment]
*decrypt its as its encrypted, create delete character from what i tested its just missing packets since when you hit create character it sends a packet (probably the check if name exists) to the server but it doesnt know what to do with it still so server doesnt respond to client and it just hangs there waiting.

right now creation/deletion and a way to pull characters from the database is needed as curently just shows the few hardcoded characters and you cant change the name.
working on that atm when i have something thats tested and working ill see if i can upload it somewhere maybe even if possible to get it in the source here.

**Comment by Kalle801** [comment]
I am able to import the unencrypted compact.sqlite to the client without errors. If that helps, let me know.

**Comment by Rain0Ash** [comment]
In archeage.exe - .xlgames? I also found it, but I think it's some kind of hash. Also I found two old programs for decrypt db, but it's not very clear how they work.  
[App.zip](https://github.com/NL0bP/Archeage-Server-emulator/files/2283043/App.zip)

**Comment by JeyFrey** [comment]
@Rain0Ash 
Its basically the same stuff that can be found in my link few comments above - you can also find a FAQ on how to use it there.
It needs a compact.sqlite3 but I only have a compact.sqlite. So I took the code and tried to exchange the sqlite3 with sqlite but I ended up with some "numbers too big" error while debugging in Visual Studio 2017.

**Comment by Rain0Ash** [comment]
@JeyFrey 
As I understand it, CA1 should create a decrypted.data file with some information in hex, however it creates an empty file. CA2 requires data in this file, but file is empty -> error. There it is said that in patch 2.0 updated the encryption base, apparently because of this.

**Comment by JeyFrey** [comment]
@Rain0Ash 
Yeah I had that same error, I just didnt knew the exact same words in english, thats why I translated it in a kinda free way. 
From what I know, you need to place the CA1 and the compact.sqlite in the unpacked game_pak folder bin32 and then start the CA1.
Did you try it that way or with a compact.sqlite only? And do you have a compact.sqlite or .sqlite3?

I seems that the method only worked before they changed the encryption OR we are just missing something.

**Comment by Rain0Ash** [comment]
@JeyFrey 
compact.sqlite, for CA1 i renamed it to compact.sqlite3, i try to add 16 hex \x00 in file head, but no difference.

**Comment by JeyFrey** [comment]
@Rain0Ash 
Guess we need to find someone who is able to decrypt it in a way that we cant see or we just need to trial and error with databases until we find the right way how they work.

**Comment by Rain0Ash** [comment]
@NL0bP It would be nice if all information about undefined packets were displayed (and save optionaly). Unfortunately, C# I can only read, but not write.

**Comment by DancerS** [comment]
I ask, create new topics for discussion on various issues, and then it's not on the topic went ... and have a long flipping to the bottom

**Comment by JeyFrey** [comment]
Can anyone explain me how the tables "accounts" and "characters" need to look like and what the syntax is?
Or can someone upload their sql database for loginserver? Struggeling to get it setup.

**Comment by NL0bP** [comment]
@JeyFrey look in the folder:
ArcheAgeLogin\bin\Debug\sql\accounts+.sql
ArcheAgeLogin\bin\Debug\sql\characters+.sql

**Comment by JeyFrey** [comment]
@NL0bP 
Thanks, its working now. As you cant accept the quest on the boat (at least I cant), how do you progress to the open world? Or what parameter is needed where to get started directly there?

**Comment by CaioFPeres** [comment]
Hello, i couldnt understand how do i make the server work. Could someone help me on this? The exe's crash after i run them...

**Comment by NL0bP** [comment]
@CaioFPeres, see https://github.com/NL0bP/Archeage-Server-emulator/releases/tag/1.2.12 or compile a project

**Comment by CaioFPeres** [comment]
Well i have seen it but the exe's crash. Do i have to create an database under 3306 port for it to run?

**Comment by NL0bP** [comment]
@CaioFPeres, all tables must be installed in the database archeage

**Comment by CaioFPeres** [comment]
Ok, i did it. But now it wont get past char creation screen. I put name and click, and nothing happens...

**Comment by CaioFPeres** [comment]
Ok, thanks for the replys. But on the past pictures you posted on the main page, there was showing the client 3.0.3.0 running into the world. Can't it do it?

**Comment by NL0bP** [comment]
@CaioFPeres, hardcoded packages can. You cannot make your character and them to enter.
